drivers: ps2: microchip: Unable to build PS2 driver with PM_DEVICE disabled

**Notes (delete this)**

**Describe the bug**
MEC15/MEC172x PS2 driver doesn't allow usage with PM_DEVICE disabled

Using MEC15xx/MEC17xx EVB and modular cards.
Try to modify board device tree to remove pinctrl sleep support.

**To Reproduce**
Steps to reproduce the behavior:
1. cd samples/drivers/ps2
2. west build -c -p always -b mec15xxevb_assy6853 -- -DCONFIG_PM_DEVICE=n
3. See error

**Expected behavior**
PS2 driver as other drivers should allow PM_DEVICE to be disabled.

**Impact**
Prevents PS2 enabling for some boards with Zephyr 3.5 release candidate.

**Logs and console output**

                 from /home/jamezaar/sandbox/ecfwwork/zephyr_fork/drivers/ps2/ps2_mchp_xec.c:10:
/home/jamezaar/ecwork/zephyr_albertofloyd/samples/drivers/ps2/build/zephyr/include/generated/devicetree_generated.h:22730:55: error: expected declaration specifiers or '...' before numeric constant
22730 | #define DT_FOREACH_OKAY_INST_microchip_xec_ps2(fn) fn(0) fn(1)
      |                                                       ^
/home/jamezaar/sandbox/ecfwwork/zephyr_fork/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/jamezaar/sandbox/ecfwwork/zephyr_fork/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/jamezaar/sandbox/ecfwwork/zephyr_fork/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/jamezaar/sandbox/ecfwwork/zephyr_fork/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/jamezaar/sandbox/ecfwwork/zephyr_fork/include/zephyr/devicetree.h:4071:9: note: in expansion of macro 'COND_CODE_1'
 4071 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \

--- a/drivers/ps2/ps2_mchp_xec.c
+++ b/drivers/ps2/ps2_mchp_xec.c
@@ -16,10 +16,8 @@
 #include <zephyr/drivers/interrupt_controller/intc_mchp_xec_ecia.h>
 #endif
 #include <zephyr/drivers/pinctrl.h>
-#ifdef CONFIG_PM_DEVICE
 #include <zephyr/pm/device.h>
 #include <zephyr/pm/policy.h>
-#endif
 #include <zephyr/drivers/ps2.h>
 #include <soc.h>
 #include <zephyr/logging/log.h>

